client/src/components/NavbarComponent.js

```diff
@@ -36,37 +35,55 @@ class NavbarComponent extends Component {
   handleLoginFormSubmit = event => {
     event.preventDefault();
     const { loginEmail: username, loginPassword: password } = this.state;
-    const user = {username, password};
+    const user = { username, password };
     API.login(user)
       // check if user has any plans
       .then(() => API.hasPlan())
-        .then(plan => {
-          if (plan) {
-            // if they do, show them
-            window.location.href="/calendar"
-          } else {
-            // if not, have them create one
-            window.location.href="/setup"
-          }
-        });
+      .catch(err => {
+        // something went wrong with login
+        // display vague error
+      })
+      .then(plan => {
+        // if they do, show them
+        window.location.href = "/calendar"
+      }).catch(err => {
+        // 404 no plan, redirect them to setup 
+        window.location.href = "/setup"
+      });
   };
 
   handleSignupFormSubmit = event => {
-    event.preventDefault();
+    event.preventDeffr78xzault();
```

Comment:
looks like a phantom keypress here.. 
Should be preventDefault I am guessing?

Reply:
yep, cat on the computer i bet